### PR TITLE
Fix protobuf bugs in service

### PR DESCRIPTION
**Pull request by LinyuYao1021**
This commit will fix the following bugs with Protobuf FileDescriptor parsing.

1. Service is missing in FileDescriptorProto of schema generated by FileDescriptorUtils.
2. Service is missing from FileDescriptorProto to schema text.

This commit is part of bug fixes reported in issue #1671

### CODE REVIEW

File: utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java [hunk 1]
```diff
@@ -374,6 +386,36 @@ private static EnumDescriptorProto enumElementToProto(EnumType enumElem) {
...
+        ServiceDescriptorProto.Builder builder = ServiceDescriptorProto.newBuilder().setName(serviceElem.name());
+
+        for (Rpc rpc : serviceElem.rpcs()) {
+            MethodDescriptorProto.Builder methodBuilder = MethodDescriptorProto
+                    .newBuilder()
+                    .setName(rpc.getName())
+                    .setInputType(getTypeName(rpc.getRequestType().toString()))
+                    .setOutputType(getTypeName(rpc.getResponseType().toString()));
+
+            Boolean deprecated = findOptionBoolean(DEPRECATED_OPTION, rpc.getOptions());
```
Comment (blacktooth) [inline utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java:399]: Can there be other options?

File: utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java [hunk 2]
```diff
@@ -374,6 +386,36 @@ private static EnumDescriptorProto enumElementToProto(EnumType enumElem) {
...
 
+    private static DescriptorProtos.ServiceDescriptorProto serviceElementToProto(Service serviceElem) {
+        ServiceDescriptorProto.Builder builder = ServiceDescriptorProto.newBuilder().setName(serviceElem.name());
+
+        for (Rpc rpc : serviceElem.rpcs()) {
+            MethodDescriptorProto.Builder methodBuilder = MethodDescriptorProto
+                    .newBuilder()
+                    .setName(rpc.getName())
+                    .setInputType(getTypeName(rpc.getRequestType().toString()))
+                    .setOutputType(getTypeName(rpc.getResponseType().toString()));
```
Comment (blacktooth) [inline utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java:397]: We need to support streaming option as well.

File: utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java [hunk 3]
```diff
@@ -537,6 +584,19 @@ private static EnumElement toEnum(EnumDescriptorProto ed) {
...
+        String name = sv.getName();
+        ImmutableList.Builder<RpcElement> rpcs = ImmutableList.builder();
+        for (MethodDescriptorProto md : sv.getMethodList()) {
+            rpcs.add(new RpcElement(DEFAULT_LOCATION, md.getName(), "", md.getInputType(),
+                    md.getOutputType(), md.getClientStreaming(), md.getServerStreaming(),
+                    getOptionList(md.getOptions().hasDeprecated(), md.getOptions().getDeprecated())));
+        }
+
+        return new ServiceElement(DEFAULT_LOCATION, name, "", rpcs.build(),
+                getOptionList(sv.getOptions().hasDeprecated(), sv.getOptions().getDeprecated()));
```
Comment (blacktooth) [inline utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java:597]: Can there be other options?

File: utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java (hunk 1 above)
Comment (LinyuYao1021) [reply utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java:399]: I checked the Service class and there's only deprecated.

File: utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java (hunk 3 above)
Comment (LinyuYao1021) [reply utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java:597]: I checked the Service class and there's only deprecated.

File: utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax2References.proto [hunk 4]
```diff
@@ -67,3 +67,8 @@ enum AddressType {
 message AnotherType {
     message Address {}
 }
+
+service Test {
```
[the remark this answers is not in the document]
Comment (LinyuYao1021) [reply utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax2References.proto:71]: Updated

File: utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax3References.proto [hunk 5]
```diff
@@ -68,3 +68,8 @@ enum AddressType {
 message AnotherType {
     message Address {}
 }
+
+service Test {
+  rpc Foo(Customer) returns(Address);
```
[the remark this answers is not in the document]
Comment (LinyuYao1021) [reply utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax3References.proto:73]: Updated

File: utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java (hunk 2 above)
Comment (LinyuYao1021) [reply utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java:397]: Updated

File: utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax3References.proto [hunk 6]
```diff
@@ -68,3 +68,27 @@ enum AddressType {
...
+service OriginalService {
+  rpc Foo(Customer) returns(Address);
+}
+
+service DeprecatedService {
+  rpc Foo(Customer) returns(Address);
+  option deprecated = true;
+}
+
+service DeprecatedMethod {
```
Comment (blacktooth) [inline utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax3References.proto:81]: Can you add some messages in between the services?

Comment (LinyuYao1021) [reply utils/protobuf-schema-utilities/src/test/proto/TestOrderingSyntax3References.proto:81]: Updated

**Review by blacktooth — APPROVED** [review]
Looks good!

**Review by carlesarnal — APPROVED** [review]
LGTM

**Pull request merged**